### PR TITLE
Adding tests for RunningJobs class

### PR DESCRIPTION
**Pull request by ironcerocloudbees**
## Add comprehensive test coverage for RunningJobs build cancellation

This PR adds complete test coverage for the `RunningJobs` class and related build cancellation functionality, which previously had 0% test coverage.

**What's included:**
- **17 unit tests** (`RunningJobsTest`) covering cancellation logic, policy validation, and edge cases
- **6 integration tests** (`RunningJobsHudsonTest`) verifying actual build cancellation in Jenkins queue and executors
- **10 tests** for interruption causes (`NewPatchSetInterruptionTest`, `AbandonedPatchsetInterruptionTest`)

**Test coverage:**
- ✅ New patchset cancellation (default behavior)
- ✅ Abort any patchset policy (including older patchsets)
- ✅ Queued build cancellation
- ✅ Abandoned patchset cancellation
- ✅ Topic-based cancellation
- ✅ Manual patchset handling
- ✅ Policy disabled scenarios
- ✅ Different changes don't interfere

This establishes baseline test coverage before the planned RunningJobs/BuildMemory unification.

### Testing done

**All tests passing:**
Tests run: 33, Failures: 0, Errors: 0, Skipped: 0

- Unit tests use Mockito to verify cancellation logic without Jenkins infrastructure
- Integration tests use `JenkinsRule` and `SshdServerMock` to verify actual build cancellation behavior
- All tests follow checkstyle rules and are fully documented

**Manual verification:**
- Tested all cancellation policies (enabled/disabled, various combinations)
- Verified builds are properly cancelled in queue and while running
- Confirmed different changes don't interfere with each other
- Validated interruption causes appear correctly in build logs

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

### CODE REVIEW

**Comment by ironcerocloudbees** [comment]
Discarded for the moment.

File: src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/AbandonedPatchsetInterruptionTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for {@link AbandonedPatchsetInterruption}.
+ *
+ * @author Ignacio Roncero &lt;ironcero@cloudbees.com&gt;
+ */
+public class AbandonedPatchsetInterruptionTest {
+
```
Comment (rsandell) [inline src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/AbandonedPatchsetInterruptionTest.java:16]: These looks irrelevant and will only waste build time.

File: src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/NewPatchSetInterruptionTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for {@link NewPatchSetInterruption}.
+ *
+ * @author Ignacio Roncero &lt;ironcero@cloudbees.com&gt;
+ */
+public class NewPatchSetInterruptionTest {
+
```
Comment (rsandell) [inline src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/NewPatchSetInterruptionTest.java:16]: therse also seems irrelevant.

File: src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/BuildCancellationIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+ * Integration tests for build cancellation feature that verify actual job cancellation
+ * in Jenkins queue and running executors.
+ *
+ * @author Ignacio Roncero &lt;ironcero@cloudbees.com&gt;
+ */
+public class BuildCancellationIntegrationTest {
+
+    /**
+     * An instance of Jenkins Rule.
+     */
```
Comment (rsandell) [inline src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/BuildCancellationIntegrationTest.java:45]: looks like good additions .

File: src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/RunningJobsTest.java [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link RunningJobs}.
+ *
+ * @author Ignacio Roncero &lt;ironcero@cloudbees.com&gt;
+ */
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class RunningJobsTest {
+
```
Comment (rsandell) [inline src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/RunningJobsTest.java:36]: I'm not sure if these are tersting anything relevant?

File: src/test/resources/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/BuildCancellationIntegrationTest/testAbandonedPatchsetCancelsRunningBuild/gerrit-trigger.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (rsandell) [inline src/test/resources/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/BuildCancellationIntegrationTest/testAbandonedPatchsetCancelsRunningBuild/gerrit-trigger.xml:1]: the config files looks to be the same, so I believe they could be just one file in ´src/test/resources/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/BuildCancellationIntegrationTest/´

It also looks like the default settings mostly so it could be added programattically perhaps?
